### PR TITLE
Add compliances subcollection

### PR DESCRIPTION
**Pull request by d-m-u**
Per the convo in the main gitter room last night about api support of compliances [February 18, 2020 4:51 PM](https://gitter.im/ManageIQ/manageiq?at=5e4c5c6b5fd3f22ede92d754) I thought I may as well go ahead and open this.

### CODE REVIEW

**Comment by d-m-u** [comment]
@abellotti any chance I could get you to take a real quick look, please? thanks.

**Comment by chessbyte** [comment]
@Real-Omar-Afifi want to try running with this PR to see if it gets you the info you want?

**Comment by Fryguy** [comment]
I think @Real-Omar-Afifi also need compliance_details, but this can be added as a start and that as a follow up.

File: app/controllers/api/vms_controller.rb [hunk 1]
```diff
@@ -1,6 +1,7 @@
 module Api
   class VmsController < BaseController
     include Subcollections::Disks
+    include Subcollections::Compliances
```
Comment (Fryguy) [inline app/controllers/api/vms_controller.rb]: Minor, but can you make alphabetical?

Comment (d-m-u) [reply app/controllers/api/vms_controller.rb]: yeah i was gonna do a follow-up for it

File: config/api.yml [hunk 2]
```diff
@@ -4107,6 +4114,11 @@
       - :name: read
         :identifier:
         - vm_show
+    :compliances_subcollection_actions:
+      :get:
+      - :name: read
+        :identifier:
+        - vm_show
```
Comment (Fryguy) [inline config/api.yml]: compliances are technically on a bunch of different resources.  @abellotti Is there a way to do an "OR" of identifiers?  Not sure what the right identifiers are, but I usually go with whatever the UI does... @himdel ?

Comment (abellotti) [reply config/api.yml]: we can specify multiple identifiers (essentially the OR) via :identifiers instead of :identifier

Comment (Fryguy) [reply config/api.yml]: Note, I'm more thinking of if this is a subcollection of more than just vms.

**Comment by abellotti** [comment]
This is great @d-m-u, thanks for adding the tests that exercises the attributes=compliance_details. 👍

**Comment by abellotti** [comment]
@d-m-u could you take care of the rubocop warnings ?  Thanks.

File: config/api.yml (hunk 2 above)
Comment (himdel) [reply config/api.yml]: There's a bunch of `*_check_compliance` features that the UI uses for the "Policy / Check Compliance of the last known configuration"  toolbar button.

So, I guess `instance_check_compliance`, `image_check_compliance`, `vm_check_compliance` and `miq_template_check_compliance` here?

File: config/api.yml [hunk 3]
```diff
@@ -4107,6 +4114,14 @@
...
         :identifier:
         - vm_show
+    :compliances_subcollection_actions:
+      :get:
+      - :name: read
+        :identifiers:
+        - :klass: Vm
+          :identifier: vm_show
+          :identifier: vm_check_compliance
+          :identifier: instance_check_compliance
```
Comment (Fryguy) [inline config/api.yml]: I don't think this is valid YAML...

Comment (Fryguy) [reply config/api.yml]:
```ruby
s = <<~YAML
       - :klass: Vm
         :identifier: vm_show
         :identifier: vm_check_compliance
         :identifier: instance_check_compliance
YAML
YAML.load(s)
# => [{:klass=>"Vm", :identifier=>"instance_check_compliance"}]
```

Notice the keys are lost.

Comment (Fryguy) [reply config/api.yml]: LOL...this is actually what rubocop found.

Comment (d-m-u) [reply config/api.yml]: yeah, we don't really handle multiple identifiers all that well anyway as it turns out

**Comment by abellotti** [comment]
@d-m-u try the following:

Move the subcollection action under the vms directly:

```
  :compliances_subcollection_actions:
    :get:
    - :name: get
      :identifier:
      - vm_show
      - vm_check_compliance
```

You can see such examples in the :vms: collection.

With this no need to class identify since it's per collection, and the :identifier: handles lists.

Thanks,
Alberto

File: config/api.yml [hunk 4]
```diff
@@ -745,6 +745,12 @@
         :identifier: ems_cluster_protect
       - :name: unassign
         :identifier: ems_cluster_protect
+  :compliances:
+    :description: Compliances
+    :options:
+    - :subcollection
+    :verbs: *g
```
Comment (chessbyte) [inline config/api.yml]: what does the value `*g` mean?

Comment (d-m-u) [reply config/api.yml]: the verb set that can be used on the subcollection, in this case just the get

File: config/api.yml (hunk 2 above)
Comment (d-m-u) [reply config/api.yml]: thanks, I'll get to `miq_template_check_compliance` in a follow-up

File: config/api.yml (hunk 4 above)
Comment (Fryguy) [reply config/api.yml]: higher-level that is like a "pointer" in YAML, and it means to insert the object referenced with the name "g".  g is defined here: https://github.com/ManageIQ/manageiq-api/blob/a62b5f3bb439ce50cc009908e76737d05d122651/config/api.yml#L14-L15

**Comment by simaishi** [comment]
Ivanchuk backport details:

```
$ git log -1
commit b5cecaef6fcd7d755fb76607b52f6004edfd743a
Author: Gregg Tanzillo <gtanzill@redhat.com>
Date:   Mon Feb 24 09:22:00 2020 -0500

    Merge pull request #742 from d-m-u/adding_subcollection_for_compliance_details

    Add compliances subcollection

    (cherry picked from commit 309e5380171fd3bc8c0b3739c2d04adf6f7e46c0)

    Fixes https://bugzilla.redhat.com/show_bug.cgi?id=1806660
```

**Pull request merged**